src/Sarif/Readers/PropertyBagConverter.cs

```diff
@@ -106,5 +81,37 @@ public override void WriteJson(JsonWriter writer, object value, JsonSerializer s
 
             writer.WriteEndObject();
         }
+
+        public static JTokenType DotNetTypeToJTokenType(Type dotNetType)
+        {
+            JTokenType jTokenType = JTokenType.None;
```

Comment:
OK, this actually works as is. Turns out I don't need to save the actual token type. All I need to save is whether the token is a string or not, so I know whether or not to add an extra pair of quotes around it when I serialize it out. Everything else just works.